File: MaxXORSubarray.c
#include <stdio.h>
#include <stdlib.h>

//this is an implementation of kadane's algorithm
int main(){
	int a[] = {1,2,3,4,5,6,7,8,9,10};
	int currentLargest = a[0];
	int xor = 0;
	int largest = 0;
	int i = 1;
	while (i<sizeof(a)/sizeof(int)){
		xor = currentLargest^a[i];
		if (currentLargest < xor){
			currentLargest = xor;
		}
		if (currentLargest > largest){
			largest = currentLargest;
		}
		i++;
	}
	printf("%d\n",largest);
	return 0;
}

File: MaxXORSubarray.py
a = [1,2,3,4,5,6,7,8,9,10]
result = 0
xor = a[0]
#an implementation of kadane's algorithm
for i in range(1,len(a)):
    xor = xor if xor^a[i]<xor else xor^a[i]
    result = xor if xor>result else result
    #result = 0 if result<0 else result

print(result)

